### PR TITLE
Update for WP Studio Integration

### PR DESCRIPTION
**Pull request by mdbitz**
WP Studio requires the usage of companion's extend flow. They will also be wanting to add custom notices. This adds base logic to differentiate WP Studio sites from Jurassic.Ninja UI created sites.

Ran a reformat on the file as the spacings were mixed, probably should have done that separately.

Need to test in depth

### CODE REVIEW

File: companion.php [hunk 1]
```diff
@@ -3,36 +3,53 @@
...
-if ( ! defined( 'IS_ATOMIC' ) || ( IS_ATOMIC && ( defined( 'IS_ATOMIC_JN' ) && IS_ATOMIC_JN ) ) ) {
+// Setup Hooks for WP.cloud
+if ( defined( 'IS_ATOMIC_JN' ) && IS_ATOMIC_JN ) {
 	add_action( 'wp_login', 'companion_wp_login', 1, 2 );
-	add_action( 'after_setup_theme', 'companion_after_setup_theme' );
-	add_action( 'admin_notices', 'companion_admin_notices' );
 	add_action( 'pre_current_active_plugins', 'companion_hide_plugin' );
+
+	if ( defined( 'IS_WP_STUDIO' ) && IS_WP_STUDIO ) {
+		// TODO : Add hooks for WP Studio sites.
```
Comment (wojtekn) [inline companion.php:29]: This is great, so we could add a custom `companion_admin_notices_studio` action to display our custom notice.

File: companion.php [hunk 2]
```diff
@@ -3,36 +3,53 @@
...
 	return true;
 }
 
 $companion_api_base_url = get_option( 'companion_api_base_url' );
 
-// Disable functionality if Atomic and not a JurassicNinja client site.
-if ( ! defined( 'IS_ATOMIC' ) || ( IS_ATOMIC && ( defined( 'IS_ATOMIC_JN' ) && IS_ATOMIC_JN ) ) ) {
+// Setup Hooks for WP.cloud
+if ( defined( 'IS_ATOMIC_JN' ) && IS_ATOMIC_JN ) {
 	add_action( 'wp_login', 'companion_wp_login', 1, 2 );
```
Comment (wojtekn) [inline companion.php:25]: Could we move this one to the next condition, too, so we can decide whether to add this feature for Studio sites?

For now, we assumed that updating a site from ZIP would extend the period, and we didn't assume that logging to the WP Admin would do the same.

File: companion.php [hunk 3]
```diff
@@ -300,24 +320,24 @@ function companion_wp_login() {
...
-			    wp_safe_redirect( '/wp-admin/site-editor.php' );
-		    }
+			if ( empty( get_option( 'big_sky_api_key' ) ) ) {
+				wp_safe_redirect( '/wp-admin' );
+			} else {
+				wp_safe_redirect( '/wp-admin/site-editor.php' );
+			}
 			exit( 0 );
 		}
 	} else {
```
Comment (wojtekn) [inline companion.php:343]: What is this part of the condition for?

File: companion.php (hunk 2 above)
Comment (mdbitz) [reply companion.php:25]: Yes I can move the wp_login hook as well

File: companion.php (hunk 3 above)
Comment (mdbitz) [reply companion.php:343]: BigSky is what the AI team is working on and they have been using Jurassic.Ninja to quickly prototype. This gave them a seamless experience of dropping into the editor vs having to do another click.

Comment (wojtekn) [reply companion.php:343]: Thanks, it's clear now.

**Comment by mdbitz** [comment]
We don't need this at the moment. closing.

**Pull request closed without merging**